src/main/java/com/github/freeclimbapi/utils/RequestVerifier.java

```diff
@@ -0,0 +1,70 @@
+package com.github.freeclimbapi.utils;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Date;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.binary.Hex;
+
+public class RequestVerifier {
+
+    public void verifyRequestSignature(String requestBody, String requestHeader, String signingSecret,
+            int tolerance) {
+        checkRequestBody(requestBody);
+        checkRequestHeader(requestHeader);
+        checkSigningSecret(signingSecret);
+        checkTolerance(tolerance);
+        SignatureInformation info = new SignatureInformation(requestHeader);
+        verifyTolerance(info, tolerance);
+        verifySignature(info, requestBody, signingSecret);
+    }
+
+    public void checkRequestBody(String requestBody) {
+        if (requestBody == "" || requestBody == null) {
+            throw new java.lang.RuntimeException("Request Body cannot be empty or null");
+        }
+    }
+
+    public void checkRequestHeader(String requestHeader) {
+        if (!requestHeader.contains("t=") || !requestHeader.contains("v1=")) {
+            throw new java.lang.RuntimeException(
+                    "Error with request header, ether it is null or an empty string or request header does not meet requirements");
```

Comment:
Lets have individual errors. The exception being thrown is the same as the `else if` condition below - making the if/else if moot. 
- error if request header is empty
- error if timestamp is not present
- error if signatures are not present

Reply:
Will change in latest commit